Airport1/Airport1/Program.cs

```diff
@@ -0,0 +1,470 @@
+{(int)OperationType.Exit} - Exit");
+
+                var operationType = (OperationType)int.Parse(Console.ReadLine());
+                switch (operationType)
+                {
+                    case OperationType.AddFlight: AddFlight(); break;
+                    case OperationType.ShowArrival: ShowArrivals(); break;
+                    case OperationType.ShowDepartures: ShowDepartures(); break;
+                    case OperationType.EditInformation: EditInformation(); break;
+                    case OperationType.SearchByFlightNumber: SearchByFlightNumber(); break;
+                    case OperationType.SearchByTime: SearchByTime(); break;
+                    case OperationType.SearchNearest: SearchNearest(); break;
+                    case OperationType.SearchByPort: SearchByPort(); break;
+                    case OperationType.SetEmergencyStatus: SetEmergencyStatus(); break;
+                    case OperationType.Exit: return;
+                    default: Console.WriteLine($"Program doesn't has operation type id {(int)operationType}"); break;
+                }
+            }
+        }
+
+        public static void AddFlight()
+        {
+            Console.WriteLine();
+            Console.WriteLine(@$"Enter flight type:
+{(int)FlightType.Arrival} - Arrival
+{(int)FlightType.Departure} - Departure");
+
+            var flightType = (FlightType)int.Parse(Console.ReadLine());
+
+            Console.WriteLine(@$"Enter flight date/time in format dd/mm/yy hh:mm");
+            var dateTime = DateTime.Parse(Console.ReadLine());
+
+            Console.WriteLine(@$"Enter flight city/port");
+            var cityPort = Console.ReadLine();
+
+            Console.WriteLine(@$"Enter airline");
+            var airline = Console.ReadLine();
+
+            Console.WriteLine(@$"Enter terminal");
+            Console.WriteLine(@"    Terminals:");
+            const int terminalCount = 26;
+            for (int i = 1; i <= terminalCount; i++)
+            {
+                var terminal = ((TerminalType)i).ToString();
+                Console.WriteLine(@$"    {i} = {terminal}");
+            }
+            var terminalType = (TerminalType)int.Parse(Console.ReadLine());
+
+            Console.WriteLine(@$"Enter flight status");
+            Console.WriteLine(@"    Flight Statuses:");
+            const int flightStatusesCount = 9;
+            for (int i = 1; i <= flightStatusesCount; i++)
+            {
+                var flightStatus = ((FlightStatusType)i).ToString();
+                Console.WriteLine(@$"    {i} = {flightStatus}");
+            }
+            var flightStatusType = (FlightStatusType)int.Parse(Console.ReadLine());
+
+            Console.WriteLine(@$"Enter gate:");
+            var gate = Console.ReadLine();
+
+            AppendFlight(flightType, dateTime, cityPort, airline, terminalType, flightStatusType, gate);
+            Console.WriteLine();
+        }
+
+        public static void ShowArrivals()
+        {
```

Comment:
I think you should take out code 114-122 in separate method and reuse it in all other methods. Because now i see copy of the same code in 147-155, 180-188, 256-264, 284-292 and so on/

Reply:
Fixed f7b39ee68dd1291f5144d00c457f22811c4f64e6